emulator/src/core/emulator.rs

```diff
@@ -394,6 +398,47 @@ impl Emulator {
     pub fn load_ram(&mut self, data: Vec<u8>, start: u16) {
         self.ram.load_vec(data, start)
     }
+
+    pub fn interrupt(&mut self, opcode: u8) -> EResult<()> {
```

Comment:
Shouldn't this method also check whether interrupts are currently enabled? (EI and DI instructions)

Reply:
and also reset the "INTE" bit after executing the instruction

Reply:
"Whenever INTE is equal to 0, the entire interrupt handling system is disabled, and no interrupts will be accepted."

"When the CPU recognizes an interrupt request from an
external device, the following actions occur:
(1) The instruction currently being executed is completed.
(2) The interrupt enable bit, INTE, is reset = O.
(3) The interrupting device supplies, via hardware, one instruction which the CPU executes."